sheet.go

```diff
@@ -707,6 +708,59 @@ func (f *File) SearchSheet(sheet, value string) []string {
 	return result
 }
 
+// RegSearchSheet provides the ability to retrieve coordinates
+// with the given worksheet name and regular expression
+// For a merged cell, get the coordinates
+// of the upper left corner of the merge area.
+// :example)
+// Search the coordinates where the numerical value in the range of "0 - 9" of Sheet 1 is described:
+//
+//    xlsx.RegSearchSheet("Sheet1", "[0-9]")
+//
+// return map[CELL:result CELL:result ...]
+func (f *File) RegSearchSheet(sheet, value string) map[string]string {
+	xlsx := f.workSheetReader(sheet)
+	result := map[string]string{}
+	name, ok := f.sheetMap[trimSheetName(sheet)]
+	if !ok {
+		return result
+	}
+	if xlsx != nil {
+		output, _ := xml.Marshal(f.Sheet[name])
+		f.saveFileList(name, replaceWorkSheetsRelationshipsNameSpaceBytes(output))
+	}
+	xml.NewDecoder(bytes.NewReader(f.readXML(name)))
+	d := f.sharedStringsReader()
+	var inElement string
+	var r xlsxRow
+	decoder := xml.NewDecoder(bytes.NewReader(f.readXML(name)))
+	for {
+		token, _ := decoder.Token()
+		if token == nil {
+			break
+		}
+		switch startElement := token.(type) {
+		case xml.StartElement:
+			inElement = startElement.Name.Local
+			if inElement == "row" {
+				r = xlsxRow{}
+				_ = decoder.DecodeElement(&r, &startElement)
+				for _, colCell := range r.C {
+					val, _ := colCell.getValueFrom(f, d)
+					_reg := regexp.MustCompile(value)
```

Comment:
The contents of functions `SearchSheet` and `RegSearchSheet` are mostly the same, we can refactor function `SearchSheet` to add a variable length parameter `reg` instead of add the new function `RegSearchSheet`, and change return type of function `SearchSheet` to `[][]string`.